source/common/quic/quic_stat_names.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "envoy/stats/scope.h"
+
+#include "common/common/thread.h"
+#include "common/stats/symbol_table_impl.h"
+
+#include "quiche/quic/core/quic_error_codes.h"
+#include "quiche/quic/core/quic_types.h"
+
+namespace Envoy {
+namespace Quic {
+
+class QuicStatNames {
+public:
+  // This class holds lazily symbolized stat names and is responsible for charging them.
+  explicit QuicStatNames(Stats::SymbolTable& symbol_table);
+
+  void chargeQuicConnectionCloseStats(Stats::Scope& scope, quic::QuicErrorCode error_code,
+                                      quic::ConnectionCloseSource source, bool is_upstream);
+
+private:
+  void incCounter(Stats::Scope& scope, const Stats::StatNameVec& names);
+
+  Stats::StatName connectionCloseStatName(quic::QuicErrorCode error_code);
+
+  Stats::StatNamePool stat_name_pool_;
+  Stats::SymbolTable& symbol_table_;
+  const Stats::StatName downstream_;
+  const Stats::StatName upstream_;
```

Comment:
does this PR do upstream stats?  Should we have a TODO to wire those up?

Reply:
Currently this is only for downstream since our downstream customers have a tighter timeline. Added TODO to plumb it through upstreams.